dali/python/backend_impl.cc

```diff
@@ -183,6 +183,8 @@ void FillTensorFromDlPack(py::capsule capsule, SourceDataType<SrcBackend> *batch
     batch->set_pinned(true);
   } else if (dl_tensor.device.device_type == kDLCPU) {
     batch->set_device_id(-1);
+    if (RestrictPinnedMemUsage())
+      batch->set_pinned(false);
```

Comment:
```suggestion
    batch->set_pinned(false);
```
`batch` just wraps the memory provided by DLPack. So it if it kDLCPU it is not be pinned.

Reply:
I didn't want to slow it down, so if it was pinned, I don't want to make it unconditionally non-pinned.

Reply:
It never was. I think it is just a bug.
kDLCUDAHost - pinned
kDLCPU - not pinned
And:
```
batch->ShareData(shared_ptr<void>(dl_tensor.data,
                                    [dlm_tensor_ptr = move(dlm_tensor_ptr)](void*) {}),
                                    bytes, typed_shape, dali_type.id());```
just accepts the data from the outside world, and all we know about it is inside `device_type`.
I think the bug here comes from the fact that our tensors are pinned by default what can be easily missed.

Reply:
But we can leave it as it is.

Reply:
OK, I'll change it.